### PR TITLE
Ease the generation of anonymized CSV datasets and delete notebooks

### PR DESCRIPTION
**Pull request by jilljenn**
Minimal PR. Main ideas:

- Allow computing the RMSE when there are `None` entries.
- Titles can be added to dataset backup files.
- Add a `save_csv` method to make anonymized datasets.

The rest is less significant (actually updating the notebooks). This PR makes me realize we should put the notebooks in another repository.

### CODE REVIEW

File: mangaki/mangaki/utils/data.py [hunk 1]
```diff
@@ -33,12 +34,33 @@ def load(self, filename):
...
         self.encode_work = backup.encode_work
         self.decode_work = backup.decode_work
         self.interesting_works = backup.interesting_works
 
+    def save_csv(self, suffix=''):
+        ratings_path = os.path.join(settings.DATA_DIR, 'ratings{}.csv'.format(suffix))
+        works_path = os.path.join(settings.DATA_DIR, 'works{}.csv'.format(suffix))
+        if os.path.isfile(ratings_path) or os.path.isfile(works_path):
+            confirm = input('Already exists. Continue? [y/n] ')
+        if confirm:
```
Comment (Elarnon) [inline mangaki/mangaki/utils/data.py]: Answering `n` to the prompt will confirm, which is probably not what you want.

Comment (jilljenn) [inline mangaki/mangaki/utils/data.py]: :DDDDDDD

File: mangaki/mangaki/utils/data.py [hunk 2]
```diff
@@ -33,12 +34,33 @@ def load(self, filename):
...
         self.encode_work = backup.encode_work
         self.decode_work = backup.decode_work
         self.interesting_works = backup.interesting_works
 
+    def save_csv(self, suffix=''):
+        ratings_path = os.path.join(settings.DATA_DIR, 'ratings{}.csv'.format(suffix))
+        works_path = os.path.join(settings.DATA_DIR, 'works{}.csv'.format(suffix))
+        if os.path.isfile(ratings_path) or os.path.isfile(works_path):
+            confirm = input('Already exists. Continue? [y/n] ') == 'y'
+        if confirm:
```
Comment (Elarnon) [inline mangaki/mangaki/utils/data.py]: `confirm` is not defined if there is not an existing file.

Comment (jilljenn) [inline mangaki/mangaki/utils/data.py]: Tu aurais pu me le dire avant >P

Comment (Elarnon) [inline mangaki/mangaki/utils/data.py]: C'est la méthode en deux étapes pour déprimer les gens.

(non en vrai j'avais pas vu)

File: mangaki/mangaki/utils/dpp2.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        D, V = np.linalg.eig(self.L[np.ix_(self.indices, self.indices)])
+        self.D = np.real(D)
+        self.V = np.real(V)
+
+    def sample_k(self, k):
+        volumes = []
+        for _ in range(REPEAT):
+            sampled = [int(index) for index in dpplib.sample_k(k, self.D, self.V)]
+            volumes.append((get_volume(self.vectors[sampled]), sampled))
+        #print(volumes)
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/dpp2.py]: Commented code.

File: mangaki/mangaki/utils/dpp2.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        self.preprocess()
+
+    def preprocess(self):
+        D, V = np.linalg.eig(self.L[np.ix_(self.indices, self.indices)])
+        self.D = np.real(D)
+        self.V = np.real(V)
+
+    def sample_k(self, k):
+        volumes = []
+        for _ in range(REPEAT):
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/dpp2.py]: `REPEAT` should be a parameter of the instance, IMHO.

File: mangaki/mangaki/utils/dpp2.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    def __init__(self, vectors=None):
+        self.vectors = vectors
+        self.items = list(range(len(vectors)))
+
+    def sample_k(self, nb_points):
+        volumes = []
+        for _ in range(REPEAT):
+            sampled = sample(self.items, nb_points)
+            volumes.append((get_volume(self.vectors[sampled]), sampled))
+        #print(volumes)
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/dpp2.py]: Commented code.

File: mangaki/mangaki/utils/dpp2.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class MangakiUniform:
+    def __init__(self, vectors=None):
+        self.vectors = vectors
+        self.items = list(range(len(vectors)))
+
+    def sample_k(self, nb_points):
+        volumes = []
+        for _ in range(REPEAT):
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/dpp2.py]: Should be a parameter of the instance.

File: mangaki/mangaki/utils/efa.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
+from mangaki.utils.common import RecommendationAlgorithm
+from django.conf import settings
+from mangaki.utils.chrono import Chrono
+from sklearn.decomposition import FactorAnalysis
+from scipy.sparse import csr_matrix
+import numpy as np
+
+
+class MangakiEFA(RecommendationAlgorithm):
```
Comment (Elarnon) [inline mangaki/mangaki/utils/efa.py]: Please add a docstring with at least a link to relevant paper / explanation of the algorithm,

Comment (Elarnon) [inline mangaki/mangaki/utils/efa.py]: (Also what is the link with this PR?)

File: mangaki/mangaki/utils/dpp2.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+from mangaki.utils.algo import get_algo_backup, get_dataset_backup
+from random import sample
+import numpy as np
+
+
+def get_volume(vectors):
+    return np.linalg.det(vectors.dot(vectors.T))
+
+
+class MangakiUniform:
```
Comment (Elarnon) [inline mangaki/mangaki/utils/dpp2.py]: What is the link with this PR?

File: mangaki/mangaki/utils/common.py [hunk 9]
```diff
@@ -44,8 +44,9 @@ def get_shortname(self):
     def get_backup_filename(self):
         return '%s.pickle' % self.get_shortname()
 
-    def compute_rmse(self, y_pred, y_test):
-        return mean_squared_error(y_pred, y_test) ** 0.5
+    def compute_rmse(self, y_pred, y_true):
```
Comment (Elarnon) [inline mangaki/mangaki/utils/common.py]: What's the link with this PR?

File: mangaki/mangaki/management/commands/compare.py [hunk 10]
```diff
@@ -34,12 +34,20 @@
 class Experiment(object):
     def __init__(self, dataset_name):
         self.algos = [
+            lambda: MangakiALS(10),
```
Comment (Elarnon) [inline mangaki/mangaki/management/commands/compare.py]: What's the link with this PR?

File: mangaki/mangaki/utils/efa.py (hunk 7 above)
Comment (jilljenn) [inline mangaki/mangaki/utils/efa.py]: Done. I will change the title.

File: mangaki/mangaki/utils/dpp2.py (hunk 8 above)
Comment (jilljenn) [inline mangaki/mangaki/utils/dpp2.py]: In order to perform benchmarks of cold-start algorithms.

**Comment by jilljenn** [comment]
- Ajouter des commentaires pour expliquer les arguments
- Mieux gérer les titres
- dpp2 wtf
- compute_rmse ne devrait pas avoir les None

**Pull request merged**